services/director-v2/src/simcore_service_director_v2/utils/dict_utils.py

```diff
@@ -32,9 +32,55 @@ def _merge_fct(a, b):
         if b_value is None:
             # skip the merge if there is no value in b
             continue
-        assert b_value is not None  # nosec
+
         merged_dict = update_in(
             merged_dict, keys_path, lambda x, b_value=b_value: _merge_fct(x, b_value)
         )
 
     return merged_dict
+
+
+def get_longest_key_paths(data: dict[str, Any]) -> tuple[list[str], ...]:
```

Comment:
Really strange name: **longest key** paths? 
if this is a tree, I would say what you are aiming is a list of key-paths of all its **leafs**

Reply:
I guess `get_leaf_key_paths` sounds better and gives an idea of what is happening.
I've also made there doctoring a bit better.